src/main/java/dev/openfeature/sdk/OpenFeatureAPI.java

```diff
@@ -286,6 +285,28 @@ public FeatureProvider getProvider(String domain) {
         return providerRepository.getProvider(domain);
     }
 
+    /**
+     * Return the state of the default provider.
+     */
+    public ProviderState getProviderState() {
+        return providerRepository.getProviderState();
+    }
+
+    /**
+     * Get the state of the provider for a domain. If no provider with the domain is found, returns the state of the
+     * default provider.
+     *
+     * @param domain The domain to look for.
+     * @return A named {@link FeatureProvider}
+     */
+    public ProviderState getProviderState(String domain) {
+        return providerRepository.getProviderState(domain);
+    }
+
+    public ProviderState getProviderState(FeatureProvider provider) {
```

Comment:
I know I've requested this, but maybe it's better to not expose this on the public API surface unless someone requests it. We could also get rid of the additional code in the `ProviderRepository` that handles this request again.

Sorry for being too greedy 🙈

Reply:
This overload would return a `null` state if the provider is not registered, which is what I would expect (or an exception?), but it's still different to the behavior of other `getProviderState()` overloads.

Reply:
Yes I guess the only reason this was done was for [this assertion](https://github.com/open-feature/java-sdk/pull/1096/files#r1768715805) which we can avoid. I agree we should remove these methods.

Reply:
Done: https://github.com/open-feature/java-sdk/pull/1096/commits/45e9ba2e0b33ee84b0449003daff3f03cb820a45